### PR TITLE
Add a usage notification

### PR DESCRIPTION
**Pull request by dlsniper**
This is the refactoring of #1357

### CODE REVIEW

**Comment by dlsniper** [comment]
@zolotov I've changed the PR as you advised. The only issue that I have right now is persisting the state as I don't understand what I'm missing to enable that one.

**Comment by zolotov** [comment]
@dlsniper as I said use PropertiesComponent.getInstance() for persistent

**Comment by zolotov** [comment]
https://github.com/dlsniper/google-go-lang-idea-plugin/blob/usage--notification/src/com/goide/project/GoModuleLibrariesInitializer.java#L263, but don't pass the project in order to save state on application level

**Comment by dlsniper** [comment]
Done. I didn't understood I should use the "getInstance()" vs implement the methods themselves, hence the confusion. Thanks for helping out.

File: src/com/goide/ui/ProjectTutorialNotification.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      if (!shownAlready) {
+        propertiesComponent.setValue(GO_PROJECT_TUTORIAL_NOTIFICATION_SHOWN, PLUGIN_VERSION);
+      }
+    }
+    
+    if (shownAlready) {
+      return;
+    }
+
+    NotificationGroup group = NotificationGroup.findRegisteredGroup("Plugins Suggestion");
```
Comment (zolotov) [inline src/com/goide/ui/ProjectTutorialNotification.java]: Still don't understand why do you need group. Why not just

```
Notifications.Bus.notify(new Notification());
```

Group will be found by its name or created.

'Plugins Suggestion' is not the best name, it's unlikely that user will be able to associate it with Go Plugin. Use 'Go'-group (it's used in quick fixes) or make the new one but with 'Go' word.

File: src/com/goide/ui/ProjectTutorialNotification.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  public void initComponent() {
+    final PropertiesComponent propertiesComponent = PropertiesComponent.getInstance();
+    boolean shownAlready;
+    //noinspection SynchronizationOnLocalVariableOrMethodParameter
+    synchronized (propertiesComponent) {
+      String storedVersion = propertiesComponent.getValue(GO_PROJECT_TUTORIAL_NOTIFICATION_SHOWN, "0.0");
+      shownAlready = VersionComparatorUtil.compare(storedVersion, PLUGIN_VERSION) == 0;
+
+      if (!shownAlready) {
+        propertiesComponent.setValue(GO_PROJECT_TUTORIAL_NOTIFICATION_SHOWN, PLUGIN_VERSION);
```
Comment (zolotov) [inline src/com/goide/ui/ProjectTutorialNotification.java]: I believe that you can set value without checking 'shownAlready'

**Comment by zolotov** [comment]
It's ok now. Please simplify notification showing a little.

File: src/com/goide/ui/ProjectTutorialNotification.java (hunk 2 above)
Comment (dlsniper) [inline src/com/goide/ui/ProjectTutorialNotification.java]: Unfortunately ```getOrInit()``` doesn't return null if the value didn't existed before.
Wouldn't it be better to write the value there only if it's not there already? If I understand correctly, please correct me if not, the value would be written every time the IDE starts up and I'd rather avoid that.

Comment (zolotov) [inline src/com/goide/ui/ProjectTutorialNotification.java]: I didn't say anything about getOrInit(), I said that ```if (!showAlready)``` seems to be redundant

Comment (dlsniper) [inline src/com/goide/ui/ProjectTutorialNotification.java]: Like I've said, wouldn't it create one more disk write on startup every time IDEA starts? I'm happy to remove the check, but I want to understand how things work better before doing so :) Thank you.

Comment (zolotov) [inline src/com/goide/ui/ProjectTutorialNotification.java]: > Like I've said, wouldn't it create one more disk write on startup every time IDEA starts?

No, you'll just add value to map. PropertiesComponent will make disk write operations no matter you added/replaced value or not.

Comment (dlsniper) [inline src/com/goide/ui/ProjectTutorialNotification.java]: I see, thanks. Updated and squashed. Thank you.

**Pull request merged**